### PR TITLE
Content for View and reset consumer group offset

### LINKED ISSUE #367: Migrating Viewing and resetting consumer group offsets Kafka Article
# What should we add to our content collections?

Migrating https://help.aiven.io/en/articles/2661525-viewing-and-resetting-consumer-group-offsets

# Is this something you would work on yourself?

[X] I plan to open a pull request for this

### PR DESCRIPTION
**Pull request by ftisiot**
# What changed, and why it matters

Migrated http://127.0.0.1:8000/docs/products/kafka/howto/viewing-resetting-offset.html 

I'm not sure about the title, but I reworked a lot the content

Fixes #367

### CODE REVIEW

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
+View and reset consumer group offsets
+============================================
+
+The `open source Apache Kafka code <https://kafka.apache.org/downloads>`_ includes a ``kafka-consumer-groups.sh`` utility enabling you to view and manipulate the state of consumer groups. 
+Before using the tool, few setup steps need to be performed.
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:5]: "few" is probably correct, but if we mean "some" - better "a few"

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Define the configuration file
+-----------------------------
+
+#. Create the Java keystore and truststore for your Aiven for Apache Kafka service using the :ref:`dedicated Aiven CLI command <avn_service_user_kafka_java_creds>`.
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:10]: We have another article explaining how to create keystore and truststore in more details! I think we should link it here  once it is merged :)

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+#. Create a ``consumer.properties`` properties file pointing to the keystore and truststore files with the following entries:
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:12]: "a ``consumer.properties`` properties file" - "a ``consumer.properties`` file" I wouldn't repeat "properties".

Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:12]: "keystore and truststore files" - to avoid saying word "file" again, can we replace it with "keystore and truststore certificates", those files are certificates, right?

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+* ``security.protocol``: security protocol, SSL for the default TLS security settings
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:14]: I have a feeling that this set of ssl properties is usuful even outside of the context of this article. They are what is needed to connect through Java application to our Kafka cluster. Does it make sense to add it as a separate article? Or, we'll do it eventually later, we had something similar in Help articles.

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    ssl.truststore.location=client.truststore.jks
+    ssl.truststore.password=changeit
+
+
+Managing consumer group offsets with ``kafka-consumer-groups.sh``
+-----------------------------------------------------------------
+
+The ``kafka-consumer-groups.sh`` tool enables to manage consumer group offsets, the following commands are available.
+
+List the active consumer groups
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:42]: I'd remove "the"...  Since it is plural, and we're not talking about specific active consumer groups, "the" in the title seems excessive... but I'm not 100% sure of the grammar side of it. Same for other titles

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+'''''''''''''''''''''''''''''''''''''''''
+
+To list the currently active consumer groups use the following command replacing the ``demo-kafka.my-project.aivencloud.com:17072`` with the Aiven for Apache Kafka service URI:
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:45]: can we use instead of  "with the Aiven for Apache Kafka " -  "with your service URI", it is easier to read and context is clear. and since we don't mention work Kafka we have no legal issues here.

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+::
+
+    kafka-consumer-groups.sh \
+        --bootstrap-server demo-kafka.my-project.aivencloud.com:17072 \
+        --command-config consumer.properties \
+        --list
+
+An example of the ``list`` option, showing two consumer groups, is the following:
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:54]: This example output is not really useful, do we need it?

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+::
+
+    kafka-consumer-groups.sh \
+        --bootstrap-server demo-kafka.my-project.aivencloud.com:17072 \
+        --command-config consumer.properties \
+        --group my-group \
+        --describe
+          
+An example of the ``describe`` option, showing the details of the ``my-group`` consumer groups, is the following:
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:74]: I'd rephrase "An example of the ``describe`` option, showing the details of the ``my-group`` consumer groups, is the following:" to use more simple language "The details of the consumer group will be printed out in the output"

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+::
+
+    kafka-consumer-groups.sh \
+        --bootstrap-server demo-kafka.my-project.aivencloud.com:17072 \
+        --command-config consumer.properties \
+        --group my-group \
+        --describe \
+        --members
+
+An example of the ``--members`` option, showing the members of the ``my-group`` consumer groups, is the following:
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:95]: I'd again rephrase it to use simple language, unless those details bring some additional value

File: docs/products/kafka/howto/viewing-resetting-offset.rst [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+    GROUP           CONSUMER-ID                                  HOST            CLIENT-ID       #PARTITIONS
+    my-group        rdkafka-a4c0a09c-8c6e-457e-bf9e-354a8e2f4bb8 /151.62.82.140  rdkafka         0
+    my-group        rdkafka-39404560-f8f2-4b0b-9518-811e2eb20074 /151.62.82.140  rdkafka         1
+
+
+Reset the offset of a consumer group
+'''''''''''''''''''''''''''''''''''''
+
+Resetting the consumer group offset might be needed when the topic parsing needs to start at a specific (non default) offset.
```
Comment (anelook) [inline docs/products/kafka/howto/viewing-resetting-offset.rst:107]: I'd simplify this as well to avoid passive voice - "Resetting the consumer group offset might be needed" - "You might want to reset  the consumer group offset when the topic parsing needs to start at a specific (non default) offset"

**Review by anelook — CHANGES_REQUESTED** [review]
Liked the article! 

Left some comments. A big ongoing theme for me is to use more simple phrases to describe things. I'd avoid passive voice and phrase sentences so they sound more of a friendly voice rather than strict formal one.

File: docs/products/kafka/howto/viewing-resetting-offset.rst (hunk 2 above)
Comment (ftisiot) [reply docs/products/kafka/howto/viewing-resetting-offset.rst:10]: Agreed, let's wait to merge that one :D

File: docs/products/kafka/howto/viewing-resetting-offset.rst (hunk 3 above)
Comment (ftisiot) [reply docs/products/kafka/howto/viewing-resetting-offset.rst:12]: well... nope, they are files enclosing the certificates, are like lockers with the keys inside

File: docs/products/kafka/howto/viewing-resetting-offset.rst (hunk 4 above)
Comment (ftisiot) [reply docs/products/kafka/howto/viewing-resetting-offset.rst:14]: Yeah it makes sense, because this needs to be done for every Kafka tool willing to connect to our Kafka... moving into a separate file

**Review by anelook — CHANGES_REQUESTED** [review]
One small comment on a title, to remove ``highlighted`` chunk from the menu

File: docs/products/kafka/howto/kafka-tools-config-file.rst [hunk 11]
```diff
@@ -0,0 +1,34 @@
+Configure ``consumer.properties`` for Apache Kafka Toolbox
```
[the remark this answers is not in the document]
Comment (ftisiot) [reply docs/products/kafka/howto/kafka-tools-config-file.rst:1]: went for the long one since it's in the howto section so I expect people going there to "perform" an action

**Review by anelook — APPROVED** [review]
